py-polars/polars/utils/_construction.py

```diff
@@ -1036,8 +1036,13 @@ def _sequence_of_dict_to_pydf(
     )
     pydf = PyDataFrame.read_dicts(data, infer_schema_length, dicts_schema)
 
-    if column_names and set(column_names).intersection(pydf.columns()):
```

Comment:
for this section, would it work to just do
```python
    columns_equal = set(pydf.columns()) == set(column_names)
    if columns_equal and not schema_overrides:
        # skip unnecessary work
        pass
    elif column_names or schema_overrides:
        pydf = _post_apply_columns(
            pydf,
            columns=column_names,
            schema_overrides=schema_overrides,
        )
    return pydf
```
?

Reply:
I like it :)